internal/mode/static/state/graph/route_common_test.go

```diff
@@ -2661,21 +2785,192 @@ func TestIsolateL7Listeners(t *testing.T) {
 					AcceptedHostnames: map[string][]string{
 						"no-match": {},
 					},
-					Attached: true,
+					Attached:     true,
+					ListenerPort: 80,
 				},
 			},
 		},
 	}
 
-	g := NewWithT(t)
-	isolateL7RouteListeners(routes, listeners)
+	routeHostnameCafeExample := []gatewayv1.Hostname{"cafe.example.com"}
+	httpListenerRoute := createHTTPRouteWithSectionNameAndPort(
+		"hr_cafe",
+		helpers.GetPointer[gatewayv1.SectionName]("http"),
+		"test",
+		routeHostnameCafeExample...,
+	)
+
+	acceptedHostnamesHTTP := map[string][]string{
+		"http": {
+			"cafe.example.com",
+		},
+	}
+
+	routeHostnameNoSectionName := []gatewayv1.Hostname{"tea.example.com", "coffee.example.com", "flavor.example.com"}
+	hrCoffeeRoute := createHTTPRouteWithSectionNameAndPort(
+		"hr_coffee",
+		nil,
+		"test",
+		routeHostnameNoSectionName...,
+	)
 
-	result := map[string][]ParentRef{}
-	for _, route := range routes {
-		result[route.Source.GetName()] = route.ParentRefs
+	hrTeaRoute := createHTTPRouteWithSectionNameAndPort(
+		"hr_tea",
+		nil,
+		"test",
+		routeHostnameNoSectionName...,
+	)
+
+	hrFlavorRoute := createHTTPRouteWithSectionNameAndPort(
+		"hr_flavor",
+		nil,
+		"test",
+		routeHostnameNoSectionName...,
+	)
+
+	acceptedHostnamesNoSectionName := map[string][]string{
+		"hr_coffee": {"coffee.example.com"},
+		"hr_tea":    {"tea.example.com"},
+		"hr_flavor": {"flavor.example.com"},
+	}
+
+	tests := []struct {
```

Comment:
Is there a test case in here that covers the issue that @bjee19 saw in the NFR tests?